forte/data/data_store.py

```diff
@@ -452,6 +466,48 @@ def _is_annotation(self, type_name: str) -> bool:
         entry_class = get_class(type_name)
         return issubclass(entry_class, (Annotation, AudioAnnotation))
 
+    def _add_entry_raw(
+        self,
+        entry_type: Type[Entry],
+        type_name: str,
+        entry: List[Any],
+    ):
+        """
+        This function add raw entry in DataStore object
+        based on corresponding type name
+         and sort them based on entry type.
+
+        Args:
+            entry_type: entry's type which decides the sorting of entry.
+            type_name: The name of type in `self.__elements`.
+            entry: raw entry data in the list format.
+
+        Raises:
+            KeyError: raised when the entry type name is not in `self.__elements`.
+
+        Returns:
+            ``tid`` of the entry.
+        """
+        if entry_type == Annotation:
+            sorting_fn = lambda s: (
+                s[constants.BEGIN_INDEX],
+                s[constants.END_INDEX],
+            )
+            try:
+                self.__elements[type_name].add(entry)
+            except KeyError:
+                self.__elements[type_name] = SortedList(key=sorting_fn)
+                self.__elements[type_name].add(entry)
+        elif entry_type in [Link, Group]:
+            try:
+                self.__elements[type_name].append(entry)
+            except KeyError:
+                self.__elements[type_name] = []
+                self.__elements[type_name].append(entry)
+        tid = entry[constants.TID_INDEX]
+        self.__tid_ref_dict[tid] = entry
```

Comment:
`self.__tid_ref_dict` only stores annotation-like entries. We will need `self.__tid_idx_dict` to keep track of non-annotation-like entries. Refer to https://github.com/asyml/forte/blob/master/forte/data/data_store.py#L198